### PR TITLE
mailio: add version 0.22.0

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **mailio/0.22.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — COMMENTED** [review]
Thanks!

As for the error, I think it might be due to missing the `transitive_headers` trait for openssl?

**Comment by toge** [comment]
@RubenRBS 
Thanks your advise.
I fixed this problem.

**Comment by RubenRBS** [comment]
The error seems to be a persistent one, will check tomorrow what's up with it

**Comment by toge** [comment]
@RubenRBS 
Thanks a lot!

**Comment by RubenRBS** [comment]
Fixed!

This happened because the first run uploaded a corrupted package. This is a problem for the current CI implementation, as before building any package, we check if it's already there and skip it if so (`UPSTREAM` status in the logs). The problem is that there's nothing in place to catch cases where a corrupted package is found. I've talked with @uilianries for ways to improve that.

The solution was to remove the corrupted package and let it rebuild, hopefully this time it does not get corrupted

File: recipes/mailio/all/conanfile.py
Comment (RubenRBS) [inline recipes/mailio/all/conanfile.py:61]: Seems like `transitive_libs=True` for openssl will be needed :)
```suggestion
        self.requires("openssl/[>=1.1 <4]", transitive_headers=True, transitive_libs=True)
```

**Review by RubenRBS — CHANGES_REQUESTED** [review]

File: recipes/mailio/all/conanfile.py
Comment (toge) [reply recipes/mailio/all/conanfile.py:61]: Thank you for your help!

**Review by toge — COMMENTED** [review]
make openssl transitive_libs

File: recipes/mailio/all/conanfile.py
Comment (RubenRBS) [reply recipes/mailio/all/conanfile.py:60]: Seems like transitive_libs is also needed here

**Review by RubenRBS — CHANGES_REQUESTED** [review]

**Comment by toge** [comment]
@RubenRBS 
Thanks a lot for your help with this PR.

cmake requirement in build_requirements has been modified as you wrote.
https://github.com/conan-io/conan-center-index/blob/d85c212e945b3272d7ec32b8c9ecf231e6cbd28f/recipes/mailio/all/conanfile.py#L73-L74

Is there anything left to do?

**Comment by RubenRBS** [comment]
> Is there anything left to do?

Did you push the cmake changes? I don't see it in the diff. Other than that, after that fixing the cmake, we would only need to figure out what's happening with the v2 pipeline 🤔

**Review by SSE4 — APPROVED** [review]

**Comment by SSE4** [comment]
@RubenRBS can you unblock? thanks

**Comment by RubenRBS** [comment]
I want to see if we can get the v2 pipeline working before merging :)

**Comment by SSE4** [comment]
> I want to see if we can get the v2 pipeline working before merging :)

@RubenRBS can you unblock? I think v2 pipeline is not mandatory to pass here. if you need to investigate, I think you can make a copy of that branch in staging CI. it's already blocked for 3 weeks...

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**
